packages/lib/src/plugin.ts

```diff
@@ -867,7 +65,7 @@ export class BaseWebPlugin {
  * @param args - Arguments to pass on to the hook.
  * @returns Non-undefined return values from the hooks.
  */
-export async function invokeHook(plugins: Map<string, BasePlugin>, hook: string, ...args: any[]) {
+export async function invokeHook(plugins: Map<string, any>, hook: string, ...args: any[]) {
```

Comment:
I added strong typing to this function so that the hook actually have to exist and the parameters are checked. That in turn uncovered a bunch of issues, you may want to have a look.

Reply:
Took me a moment to read through the new type but it looks correct to me. I'm amazed all those uncovered issues never caused any issues or failed any tests...